.github/workflows/codechecker.yaml

```
@@ -0,0 +1,53 @@
+name: CodeChecker
+
+on: [push, pull_request]
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  codechecker_static_analyzer:
+    runs-on: ubuntu-24.04
+
+    steps:
+    - name: Checkout code including full history and submodules
+      uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+      with:
+        submodules: true
+        fetch-depth: 0
+
+    - name: Install CodeChecker
+      run:  |
+        sudo apt-get update
+        sudo apt-get install clang-tools-18 cmake cppcheck libcunit1-dev ninja-build
+        pip3 install codechecker
```

Would lock down all dependencies
```suggestion
        pip install --require-hashes -r tools/requirements-ci.txt
```

Warning: CodeChecker has quite some dependencies, this will be a rabbit hole. But maybe you'll find good tooling for this work?